Src/GoogleApis.Auth/OAuth2/AuthorizationCodeInstalledApp.cs

```diff
@@ -64,7 +64,7 @@ public ICodeReceiver CodeReceiver
 
             // If the stored token is null or it doesn't have a refresh token and the access token is expired we need 
             // to retrieve a new authorization code.
-            if (token == null || (token.RefreshToken == null && token.IsExpired(flow.Clock)))
+            if (token == null || Flow.ShouldForceTokenRetrieval() || (token.RefreshToken == null && token.IsExpired(flow.Clock)))

```

Comment:
Done. I assumed that the token == null was most likely to be true, hence the provided order.